utils/utils.go

```diff
@@ -67,9 +67,14 @@ func RunUnderSystemdScope(pid int, slice, unitName string, properties ...systemd
 		properties = append(properties, systemdDbus.PropSlice(slice))
 	}
 	ch := make(chan string)
-	_, err = conn.StartTransientUnit(unitName, "replace", properties, ch)
-	if err != nil {
-		return err
+	for {
+		_, err = conn.StartTransientUnit(unitName, "replace", properties, ch)
+		if err == nil {
+			break
+		}
+		if !errors.Is(err, syscall.EAGAIN) {
+			return err
+		}
 	}
 	defer conn.Close()
```

Comment:
Shouldn't this defer be moved earlier (to right after the error check from `systemdDbus.New()`)?